src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -979,6 +936,65 @@ protected int getNextClientIdNumber() {
         return 0;
     }
 
+    /**
+     * Create AwsIotMqttConnectionBuilder to generate an mqtt connection client.
+     *
+     * @param deviceConfiguration used to get the values needed for mqtt connection builder
+     * @param clientBootstrap client bootstrap values for mqtt connection builder
+     */
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    public AwsIotMqttConnectionBuilder createMqttConnectionBuilder(DeviceConfiguration deviceConfiguration,
+                                                                   ClientBootstrap clientBootstrap) {
+        AwsIotMqttConnectionBuilder builder;
+        try {
+            builder = securityService.getDeviceIdentityMqttConnectionBuilder();
+        } catch (MqttConnectionProviderException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Suggested remediation:
Consider passing the exception object either as  &#34;rootCause&#34; or as an inner exception parameter, to the constructor of the new exception before throwing new exception. This will help keep stack trace.
```
@@ -951,3 +951,3 @@
         } catch (MqttConnectionProviderException e) {
-            throw new MqttException(e.getMessage());
+            throw new MqttException(e.getMessage(), e);
         }

```